website/docs/chapter-04/02_teardowns.md

```diff
@@ -0,0 +1,196 @@
+
+In Robot Framework, you can define a default Test|Task Teardown in the `*** Settings ***` section of the suite using the `Test Teardown`|`Task Teardown` setting.
+This default teardown will be applied to all tests|tasks within the suite unless overridden.
+
+Individual tests|tasks can override the default teardown by specifying their own `[Teardown]` setting within the test|task.
+If you want to disable the teardown for a specific test|task, you can set `[Teardown]    NONE`, which effectively means that no teardown will be executed for that test|task.
+
+It is recommended to define the local `[Teardown]` setting as the last line of the test|task.
+
+**Key characteristics of Test|Task Teardown:**
+- Test|Task Teardown is a single keyword call with potential argument values.
+- Executed after the test|task has been executed, regardless of its status.
+- Runs even if the Test|Task Setup fails.
+- If the Test|Task Teardown fails, the test|task is marked as failed in reports and logs.
+- All keywords within the Test|Task Teardown are executed, even if one of them fails.
+- Can be set globally for all tests|tasks in a suite and overridden locally.
+
+**Typical use cases:**
+- Logging out of an application after a test|task completes.
+- Deleting test data created during the test|task.
+- Restoring configurations altered during the test|task.
+- Distinguishing phases of a test|task in *setup* (aka *preparation* or *precondition checking*), *steps*, and *teardown* (aka *clean up* or *postconditions*).
+
+
+Example of defining a default Test|Task Teardown in the suite settings:
+
+```robotframework
+*** Settings ***
+Test Teardown    Logout User    # Default Teardown for all tests
+
+
+*** Test Cases ***
+Test with Default Teardown    # Default Teardown is applied
+    Login User
+    Do Some Testing
+
+Another Test with Default Teardown    # Default Teardown is applied
+    Login User
+    Do Some other Testing
+
+Custom Teardown Test
+    Perform Test Steps
+    [Teardown]    Cleanup Specific Data    # Override the default teardown
+
+No Teardown Test
+    Perform Other Steps
+    [Teardown]    NONE    # Override and disable the teardown by case-sensitive NONE
+```
+
+
+
+## 4.2.3 Keyword Teardown
+
+::::lo[Learning Objectives]
+
+:::K1[LO-4.2.3]
+
+Recall key characteristics, benefits, and syntax of Keyword Teardown
+
+:::
+
+::::
+
+A **Keyword Teardown** is executed after a user keyword body has been executed.
+It allows for cleanup actions specific to that keyword,
+ensuring that any resources used within the keyword are properly released independently of failed child keyword calls.
+
+For better readability, it should be written as the last line of a keyword.
+
+**Key characteristics of Keyword Teardown:**
+- Keyword Teardown is a single keyword call with potential argument values.
+- Executed after the keyword body has been executed, regardless of its status.
```

Comment:
- If the Keyword Teardown fails, the test|task is marked as failed in reports and logs.

Reply:
no. see setup